### DISCUSSION TITLE
Hostname instead of IP Address

### DISCUSSION THREAD

**Discussion by daexpert** [dawidchyrzynski/arduino-home-assistant#193] [General]
How can I use hostname instead of ip address for the broker? on this line:

#define BROKER_ADDR     IPAddress(xxx.com)

I receive the following error if I replace ip with hostname:

/home/paul/snap/arduino/current/Arduino/HA_binary_sensor/HA_binary_sensor.ino: In function 'void setup()':
HA_binary_sensor:5:35: error: 'xxx' was not declared in this scope
 #define BROKER_ADDR     IPAddress(xxx.com)

**Comment by shedokan** [comment]
You need to put it as a string(aka. `char` array):
```
const char* HA_MQTT_BROKER_ADDR = "homeassistant.local";
```

And then when you `begin()` pass it as a string(I have all my parameters as constants:
```
  haMQTT.begin(HA_MQTT_BROKER_ADDR, HA_MQTT_BROKER_USER, HA_MQTT_BROKER_PASSWORD);
```
